contracts/governed_token_wrapper/lib.rs

```
@@ -454,31 +496,58 @@ mod governed_token_wrapper {
             mint_for: AccountId,
             cost_to_wrap: Balance,
             leftover: Balance,
-        ) {
+        ) -> Result<()> {
             if token_address.is_none() {
                 // mint the native value sent to the contract
-                self.mint(mint_for, leftover);
+                if self.mint(mint_for, leftover).is_err() {
+                    ink_env::debug_println!("error occured minting for native token");
```

If there is an error don't we want to return an error?